### DISCUSSION TITLE
Getting quite slow with large Data set

### DISCUSSION THREAD

**Discussion by techhysahil** [xyflow/xyflow#3101] [General]
I have about ~6k nodes that are connected using edges. I have feed the data to **Dagre Tree** [example](https://reactflow.dev/docs/examples/layout/dagre/). Now output looks like this : https://codesandbox.io/s/bold-goldstine-cks

I have 2 problems with it, any help would be appreciated. 
1. It's getting quite slow. 
2. I want them to arrange like this: node with the highest number of links should be shown first and so on.

**Comment by bcakmakoglu** [comment]
The sandbox link doesn't work.
Other than that, React Flow really isn't made to render 6k nodes at once, that's just a limitation of the HTML based approach here.
If you can't work around using that many nodes at once, you might want to consider something WebGL based or similar.

**Comment by techhysahil** [comment]
Link: https://codesandbox.io/s/bold-goldstine-cks4jt?file=/index.js
@bcakmakoglu , thanks for replying. I understand the 6k concurrent DOM nodes limitation, just wanted to know you guys viewpoint.  

Have a nice day.
